docs/reference/search/search-your-data/search-template.asciidoc

```diff
@@ -630,3 +630,414 @@ POST _render/template
   }
 }
 ----
+[[search-template-with-mustache-examples]]
+=== Search template examples with Mustache
+
+The mustache templating language defines various tag types you can use within templates. The following sections describe some of these tag types and provide examples of using them in {es} <<search-template, search templates>>.
+
+[discrete]
+[[search-template-mustache-variable]]
+=== Mustache variables
+Mustache tags are typically enclosed in double curly brackets, A mustache variable : `{{my-variable}}` is a type of mustache tag which checks for a matching `my-variable` key in your search template `params` and replaces template with the value set for `my-variable`
```

Comment:
```suggestion
Mustache tags are typically enclosed in double curly brackets, A mustache variable : `{{my-variable}}` is a type of mustache tag which checks for a matching `my-variable` key in your query `params` and replaces template with the value set for `my-variable`
```

Reply:
Maybe it's better to say "query params" here?

Reply:
I think just `params` would be fine? Similar to how current search template documentation refers to `params` in [create search template](https://www.elastic.co/guide/en/elasticsearch/reference/current/search-template.html#create-search-template)

So something like :
``` 
... checks for a matching `my-variable` key in your `params`...
```

My reasoning: `params` is optional tag set outside `query` but within `source` in search template.  WDYT?